File: android/lib/src/main/java/org/elastos/sdk/elephantwallet/contact/internal/FriendInfo.java
package org.elastos.sdk.elephantwallet.contact.internal;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.annotations.SerializedName;

public class FriendInfo extends HumanInfo {
    public class FriendJson extends HumanJson {
        @SerializedName(JsonKey.Alias)
        String alias;
    }

    public String alias;

    public int fromJson(HumanJson info) {
        if(info instanceof FriendJson == false) {
            return -1;
        }

        int ret = super.fromJsonObj(info);
        if(ret < 0) {
            return ret;
        }

        this.alias = ((FriendJson)info).alias;

        return 0;
    }

    public int fromJson(String info) {
        FriendJson json = new Gson().fromJson(info, FriendJson.class);

        int ret = fromJson(json);
        if(ret < 0) {
            return ret;
        }

        return 0;
    }

    public String toJson() {
        FriendJson json = new FriendJson();

        int ret = super.toJsonObj(json);
        if(ret < 0) {
            return null;
        }

        json.alias = this.alias;

        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        String val = gson.toJson(json);
        return val;
    }
} // class FriendInfo

File: ios/sdk/swift/internal/FriendInfo.swift

public class FriendInfo: HumanInfo {
  public class FriendJson: HumanInfo.HumanJson {
    var alias: String? = nil
    
    enum CodingKeys : String, CodingKey {
      case alias
      
      var rawValue: String {
        get {
          switch self {
            case .alias: return JsonKey.Alias
          }
        }
      }
    }
  }

  public var alias: String?

  public func fromJson(info: HumanJson) -> Int {
    if info is FriendJson == false {
      return -1
    }

    let ret = super.fromJsonObj(info: info)
    if ret < 0 {
        return ret
    }

    self.alias = (info as! FriendJson).alias

    return 0
  }

  public func fromJson(info: String) -> Int {
    let decode = try! JSONDecoder().decode(FriendJson.self, from: info.data(using: .utf8)!)
        
    let ret = fromJson(info: decode)
    if ret < 0 {
      return ret
    }

    return 0
  }

  public func toJson() -> String? {
    let json = FriendJson()

    let ret = super.toJsonObj(json: json)
    if(ret < 0) {
      return nil
    }

    json.alias = self.alias

    let encoder = JSONEncoder()
    encoder.outputFormatting = .prettyPrinted
    let encode = try! encoder.encode(json)
    let val = String(data: encode, encoding: .utf8)!
    return val
  }
} // class FriendInfo

